### DISCUSSION TITLE
CCA affected by sparsity

### DISCUSSION THREAD

**Discussion by danielcgingerich** [stuart-lab/signac#1158] [Q&A] [answered]
In the diagonal cca, I have a question regarding the initial step in the procedure: 
obtaining the correlation matrix
K = t(X) %*% Y 
where X and Y are the scaled and centered gene activity/scRNA counts matrices. 

What about the effect of sparsity on correlation?  [This stack exchange post](https://stats.stackexchange.com/questions/246788/is-correlation-as-a-metric-in-clustering-affected-by-sparsity) illustrates what can happen with sparse data correlations. 

 Sparsity seems to be a common problem when trying to infer gene networks and peak coaccessibility (cicero mentions it in the paper).  I never thought about the problem with regards to CCA, however.  I would like to know if you all had any thoughts on this while developing the method. 

Would dropout imputation on the gene activity and scRNA matrix before applying CCA improve the results?

**Comment by timoast** [comment, marked as answer]
I think it's affected to a similar extent that PCA is when running PCA on a scaled and centered scRNA-seq gene expression matrix. I haven't explored using imputation methods but you could certainly try it and see whether you think the results improve.
